lib/castle/commands/authenticate.rb

```diff
@@ -3,13 +3,18 @@
 module Castle
   module Commands
     class Authenticate
-      include WithContext
+      def initialize(context)
+        @context = context
+      end
 
       def build(options = {})
         validate!(options)
-        build_context!(options)
+        context = ContextMerger.call(@context, options[:context])
+        context = ContextSanitizer.call(context)
 
-        Castle::Command.new('authenticate', options, :post)
+        Castle::Command.new('authenticate',
+                            options.merge(context: context, sent_at: Time.now.iso8601),
```

Comment:
`Time.now.utc.is8601(3)` also shouldn't this be the responsibility of `Castle::Command` will DRY it up and add responsibility to the Object making the request